File: v1/src/main/java/com/api/v1/mappers/PurchaseResponseMapper.java
package com.api.v1.mappers;

import com.api.v1.domain.entities.Purchase;
import com.api.v1.dtos.responses.PurchaseResponseDto;

public class PurchaseResponseMapper {

    public static PurchaseResponseDto map(Purchase purchase) {
        return new PurchaseResponseDto(
                BookResponseMapper.map(purchase.book()),
                CustomerResponseMapper.map(purchase.customer()),
                purchase.bookPrice(),
                purchase.state(),
                purchase.saleTax(),
                purchase.finalPrice(),
                purchase.createdAt()
        );
    }

}

File: kotlin/v2/src/main/kotlin/com/api/v2/purchases/utils/PurchaseResponseMapper.kt
package com.api.v2.purchases.utils

import com.api.v2.book.domain.BookRepository
import com.api.v2.book.utils.BookResponseMapper
import com.api.v2.customer.domain.CustomerRepository
import com.api.v2.customer.utils.CustomerResponseMapperUtil
import com.api.v2.purchases.dtos.PurchaseResponseDto
import com.api.v2.purchases.domain.Purchase
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

@Component
class PurchaseResponseMapper {

    @Autowired
    private lateinit var customerRepository: CustomerRepository

    @Autowired
    private lateinit var bookRepository: BookRepository

    suspend fun map(purchase: Purchase): PurchaseResponseDto {
        return PurchaseResponseDto(
                purchase.orderNumber,
                CustomerResponseMapperUtil.map(customerRepository.findById(purchase.customer)!!),
                BookResponseMapper.map(bookRepository.findById(purchase.book)!!),
                purchase.price,
                purchase.finalPrice
        )
    }

}